pkg/network/network.go

```diff
@@ -101,14 +101,12 @@ func Render(conf *opv1alpha1.NetworkAddonsConfigSpec, manifestDir string, opensh
 	o := []*unstructured.Unstructured{}
 	var err error
 
-	if conf.KubeMacPool != nil {
-		// render kubeMacPool
-		o, err = renderKubeMacPool(conf, manifestDir)
-		if err != nil {
-			return nil, err
-		}
-		objs = append(objs, o...)
+	// render kubeMacPool
+	o, err = renderKubeMacPool(nil, conf, manifestDir)
```

Comment:
This is not terrible, but it breaks the logic of render. renderMultus should "render" the component, that's all. The decision whether render it or not is up to the higher layer - Render of the whole Spec.

I also don't see why is this needed for this deletion change. If you want to render objects to remove, you can reuse the current renderKubeMacPool method, except you would give it the `prev` config instead of the current one. You can use renderKubeMacPool at its current form. once with `prev` (for ObjsToRemove) once with `conf` (in this function).